docs/usage/restic/index.rst

```diff
@@ -0,0 +1,94 @@
+========================
+Restic-based replication
+========================
+
+
+
+.. sidebar:: Contents
+
+   .. contents:: Restic-based replication
+
+This document covers the design of the restic-based data mover.
+
+.. contents::
+   :depth: 2
+
+Overview
+========
+
+Restic is backup program that can back up your files
+from Linux, BSD, Mac and Windows to many different storage types, including self-hosted
+and online services. Read more about `Restic <https://restic.net/>`_
+
+Restic based data mover uses ``restic-config`` to read confgirations details for backup and restore.
+
+ResticConfig
+============
+
+.. code::  yaml
+
+   apiVersion: v1
+   kind: Secret
+   metadata:
+   name: restic-config
+   type: Opaque
+   stringData:
+      # The repository url
+      RESTIC_REPOSITORY: s3:http://minio.minio.svc.cluster.local:9000/restic-repo
+      # The repository encryption key
+      RESTIC_PASSWORD: my-secure-restic-password
+      # ENV vars specific to the back end
+      # https://restic.readthedocs.io/en/stable/030_preparing_a_new_repo.html
+      AWS_ACCESS_KEY_ID: access_key_id
+      AWS_SECRET_ACCESS_KEY: password
+
+ReplicationSoucreResticSpec
+===========================
+
+.. code:: yaml
+   
+   ---
+   apiVersion: scribe.backube/v1alpha1
+   kind: ReplicationSource
+   metadata:
+   name: restic-source
+   namespace: source
+   spec:
+      sourcePVC: data-vol
+      trigger:
+         schedule: "*/10 * * * *"
+      restic:
+         resticConfig: "restic-config"
+         copyMethod: Snapshot
+         resticArg: "backup"
```

Comment:
@JohnStrunk If it is a source it will do backup and if it's a destination it will do restore. Can both the dest and source do prune?

Reply:
While there's nothing stopping us from doing a prune in the Destination, I would expect the Source to do it. In the case of Restic, there would only be a Source most of the time. The only reason to create a Destination is if you need to actually restore the data. I don't think it's well suited to replication use-cases.

Please take a look at #111

Reply:
In that case I am keeping `resticArg` in ReplicationSource spec. Regarding the destination, I followed this
```
I'm assuming a 1:1 correspondence between restic repo and volume. Under that model, I don't think we'd gain anything by making a new CR.

_Originally posted by @JohnStrunk in https://github.com/backube/scribe/pull/111#discussion_r573917048_
```
Should we go with just adding a new type `ReplicationSourceResticSpec` CR and no destination type?

Reply:
Instead of having a "resticArg" in the CR, why not specify how often to prune
https://github.com/backube/scribe/pull/111/files#diff-ecded3b16e315d0f93bce5cb535dd21c22c828bb442996ae60825fb281afb0e3R72

My approach is #111 was to put as much of the config in the CR as possible and only have the credentials stuff in the restic secret (`.spec.restic.repository`)

----
> Should we go with just adding a new type `ReplicationSourceResticSpec` and no destination type?

We need [something for restore](https://github.com/backube/scribe/pull/111/files#diff-ecded3b16e315d0f93bce5cb535dd21c22c828bb442996ae60825fb281afb0e3R119-R131), at least to specify where the data should come from.